tilelang/language/v2/builder.py

```
@@ -0,0 +1,663 @@
+                except Exception:
+                    pass
+            value = ForwardRef(value, is_argument=True, is_class=False)
+        hints[name] = _eval_type(value, globalns=globalns, localns=localns, type_params=type_params)
+    return hints
+
+
+def _is_static_annot(annot: Any) -> bool:
+    return isinstance(annot, (dt.dtype, Buffer, Var))
+
+
+def prim_func(func: Callable[_P, _T] = None,
```

Ruff (0.14.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)